packages/site/cms-site/src/sitePreview/SitePreviewApiHelper.ts

```diff
@@ -0,0 +1,42 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+type Scope = Record<string, unknown>;
+type GraphQLClient = {
+    setHeader(key: string, value: string): unknown;
+    request(document: string, variables?: unknown): Promise<{ isAllowedSitePreview: boolean }>;
+};
+export type SitePreviewParams = {
+    scope: Scope;
+    path: string;
+    settings: {
+        includeInvisible: boolean;
+    };
+};
+
+export async function getValidatedSitePreviewParams(
+    req: NextApiRequest,
+    res: NextApiResponse,
+    graphQLClient: GraphQLClient,
+): Promise<SitePreviewParams> {
+    return {
+        scope: await getValidatedScope(req, res, graphQLClient),
+        path: req.query.path as string,
+        settings: JSON.parse(req.query.settings as string),
+    };
+}
+
+async function getValidatedScope(req: NextApiRequest, res: NextApiResponse, graphQLClient: GraphQLClient): Promise<Scope> {
+    const scope = JSON.parse(req.query.scope as string);
```

Comment:
This will fail if `req.query.scope` is undefined:

<img width="544" alt="image" src="https://github.com/vivid-planet/comet/assets/48853629/0c0533c5-4a6a-4cec-a8c5-d25e57658430">

so the if in the next line is presumably not necessary 🤔